torch/distributed/tensor/_op_schema.py

```diff
@@ -390,7 +390,7 @@ def gen_fake_args(self) -> ArgsType:
             to run the local tensor operator and get the output spec.
         """
         return tree_map_only(
-            DTensorSpec, _rebuild_tensor_from_dtensor_meta, self.args_schema
+            DTensorSpec, _rebuild_tensor_from_dtensor_meta, self.args_schema, is_leaf = lambda x: isinstance(x, DTensorSpec)
```

Comment:
why do we need these?

Reply:
This is because when we register DTensorSpec as constant in pytree, we treat it as empty container hence tree_map will just ignore DTensorSpec as valid type to map over. This is consistent with how Jax.tree_map does i think. 
```
>>> stuff2 = [Point(4, 4), Point(4,4)]
>>> jax = tree_map(lambda x: x.a + 4, stuff2)
>>> jax[0].a # returns 4
```
cc: @zou3519

Reply:
hmm I guess I would have thought that we would need some API like `pytree.register_leaf(DTensorSpec)`, instead of needing to update each call-site of `tree_map` (@zou3519 does that sound wrong?)